src/Mvc/View/Engine/Twig.php

```diff
@@ -16,19 +16,24 @@ class Twig extends Engine implements EngineInterface
     /**
      * @var \Twig_Environment
      */
-    protected $twig;
+    protected $_twig;
 
     /**
      * @param mixed|\Phalcon\Mvc\ViewBaseInterface $view
      * @param mixed|\Phalcon\DiInterface $dependencyInjector
-     * @param array $options
+     * @param mixed|\Twig_Environment $environment
      */
-    public function __construct($view, $dependencyInjector, array $options = [])
+    public function __construct($view, $dependencyInjector, $environment = null, $options = [])
```

Comment:
Which other $environment can be entered into this method???

Reply:
Right now is not possible to retrieve the environment to register functions, filters or extensions. This way the engine class is not responsible for the instantiation of the environment, thats is, you should create the environment outside, setup as you like and then pass it as a dependency to the engine via constructor.

Reply:
@gabriel-schneider what you think about $options could be changed to $twigOptions? If you see, passing $options, the class will be instantiable with each options twig. Like, don't need to pass $environment object, already exists others projects using it.

Reply:
I agree, actually there is no need to pass a environment class since there is only one. The loaders can be something to think about it since there is two of them, `Twig_Loader_Array` and `Twig_Loader_Filesystem`.
I changed the code so there is only a addition of a function to retrieve the twig environment to manipulate it, should be enough.